near-sdk-macros/src/lib.rs

```diff
@@ -43,6 +43,32 @@ struct NearMacroArgs {
     inside_nearsdk: Option<bool>,
 }
 
+fn has_nested_near_macros(item: TokenStream) -> bool {
+    syn::parse::<syn::Item>(item)
+        .ok()
+        .and_then(|item_ast| {
+            let attrs = match item_ast {
+                syn::Item::Struct(s) => s.attrs,
+                syn::Item::Enum(e) => e.attrs,
+                syn::Item::Impl(i) => i.attrs,
+                _ => vec![], // Other cases don't support near macros anyway
+            };
+
+            attrs.into_iter().find(|attr| {
+                let path_str = attr.path().to_token_stream().to_string();
+                path_str == "near" || path_str == "near_bindgen"
+            })
+        })
+        .is_some()
+}
+
+fn check_duplicate_contract_state() -> bool {
+    static CONTRACT_STATE_DEFINED: ::std::sync::atomic::AtomicBool =
+        ::std::sync::atomic::AtomicBool::new(false);
+
+    CONTRACT_STATE_DEFINED.swap(true, ::std::sync::atomic::Ordering::SeqCst)
```

Comment:
looks like lowering this to https://doc.rust-lang.org/nightly/std/sync/atomic/enum.Ordering.html#variant.AcqRel will not affect correctness

Reply:
Agree, not sure if it changes anything though:)